src/line_agg.rs

```diff
@@ -135,7 +135,11 @@ impl<K, C> Logic<K, C> {
     }
 }
 
-impl<T, K, C> LineAgg<T, K, C> {
+impl<T, K, C> LineAgg<T, K, C>
+where
+    T: Stream<Item = (K, Bytes, C)> + Unpin,
+    K: Hash + Eq + Clone,
+{
```

Comment:
This might not be necessary, or if really needed it's better to move the `new` fn to the lower block where generic contraints are explicit already.